wrangler/src/@utils/stellar-sdk-utils.js

```diff
@@ -1,84 +1,161 @@
+import BigNumber from "bignumber.js";
 import { clone } from "lodash";
-import { Keypair, Transaction } from "stellar-base";
+import { Keypair, Transaction, Networks } from "stellar-base";
 /**
- * @namespace Utils
+ * Verifies if a transaction was signed by the given account id.
+ *
+ * @function
+ * @memberof Utils
+ * @param {Transaction} transaction
+ * @param {string} accountID
+ * @example
+ * let keypair = Keypair.random();
+ * const account = new StellarSdk.Account(keypair.publicKey(), "-1");
+ *
+ * const transaction = new TransactionBuilder(account, { fee: 100 })
+ *    .setTimeout(30)
+ *    .build();
+ *
+ * transaction.sign(keypair)
+ * Utils.verifyTxSignedBy(transaction, keypair.publicKey())
+ * @returns {boolean}.
  */
-export var Utils;
-(function (Utils) {
-    /**
-     * Verifies if a transaction was signed by the given account id.
-     *
-     * @function
-     * @memberof Utils
-     * @param {Transaction} transaction
-     * @param {string} accountID
-     * @example
-     * let keypair = Keypair.random();
-     * const account = new StellarSdk.Account(keypair.publicKey(), "-1");
-     *
-     * const transaction = new TransactionBuilder(account, { fee: 100 })
-     *    .setTimeout(30)
-     *    .build();
-     *
-     * transaction.sign(keypair)
-     * Utils.verifyTxSignedBy(transaction, keypair.publicKey())
-     * @returns {boolean}.
-     */
-    function verifyTxSignedBy(transaction, accountID) {
-        return gatherTxSigners(transaction, [accountID]).length !== 0;
+export function verifyTxSignedBy(transaction, accountID) {
+  return gatherTxSigners(transaction, [accountID]).length !== 0;
+}
+
+/**
+ *
+ * gatherTxSigners checks if a transaction has been signed by one or more of
+ * the given signers, returning a list of non-repeated signers that were found to have
+ * signed the given transaction.
+ *
+ * @function
+ * @memberof Utils
+ * @param {Transaction} transaction the signed transaction.
+ * @param {string[]} signers The signers public keys.
+ * @example
+ * let keypair1 = Keypair.random();
+ * let keypair2 = Keypair.random();
+ * const account = new StellarSdk.Account(keypair1.publicKey(), "-1");
+ *
+ * const transaction = new TransactionBuilder(account, { fee: 100 })
+ *    .setTimeout(30)
+ *    .build();
+ *
+ * transaction.sign(keypair1, keypair2)
+ * Utils.gatherTxSigners(transaction, [keypair1.publicKey(), keypair2.publicKey()])
+ * @returns {string[]} a list of signers that were found to have signed the transaction.
+ */
+export function gatherTxSigners(transaction, signers) {
+  const hashedSignatureBase = transaction.hash();
+  const txSignatures = clone(transaction.signatures);
```

Comment:
Excessive `clone` call – `transaction.signatures` (as well as `txSignatures` varaible) should not be modified anywhere. XDR structures contain a lot of auxiliary garbage, so cloning signatures adds unnecessary memory consumption.